core/client/src/main/java/alluxio/client/file/policy/BlockLocationPolicy.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.annotation.PublicApi;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.util.CommonUtils;
+import alluxio.wire.WorkerNetAddress;
+
+/**
+ * <p>
+ * Interface for determine the Alluxio worker location to serve a block write or UFS block read.
+ * </p>
+ *
+ * <p>
+ * A policy must have an empty constructor to be used as default policy.
+ * </p>
+ */
+@PublicApi
+public interface BlockLocationPolicy {
+
+  /**
+   * The factory for the {@link BlockLocationPolicy}.
+   */
+  class Factory {
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory for creating {@link BlockLocationPolicy}.
+     *
+     * @param policyClassNameWithShard the block location policy class name. Optionally, it can also
+     *        contains the number of shards that a block's traffic can be sharded. For example:
+     *        alluxio.client.file.policy.DeterministicHashPolicy,
+     *        alluxio.client.file.policy.DeterministicHashPolicy@5,
```

Comment:
If the sharding factor is not constant between clients, will this policy still have the intended effects? (ie. what if a misbehaving client decides to use shard factor = # workers)